presto-main/src/main/java/com/facebook/presto/operator/aggregation/noisyaggregation/NoisySumState.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.noisyaggregation;
+
+import com.facebook.presto.operator.aggregation.state.InitialBooleanValue;
+import com.facebook.presto.spi.function.AccumulatorStateMetadata;
+import io.airlift.slice.SliceInput;
+import io.airlift.slice.SliceOutput;
+
+import static io.airlift.slice.SizeOf.SIZE_OF_BYTE;
+import static io.airlift.slice.SizeOf.SIZE_OF_DOUBLE;
+
+@AccumulatorStateMetadata(stateSerializerClass = NoisySumStateSerializer.class)
+public interface NoisySumState
```

Comment:
conceptually I find it a bit confusing that the sum state extends the count state though I don't have a much better proposal (apart from maybe renaming NoisyCountState to something more generic like NoisyAggState but then it might be confusing what the connection to count is)

Reply:
perhaps add comments to what NoisyCountState represents in the corresponding file

Reply:
Thanks, I will add more docstring to the class.
In the implementation of normal SUM, there is a boolean flag of isNull which basically means there is no rows. If I understand correctly, it is equivalent to storing count explicitly and check if that count == 0. Maybe there is other context that I dont know

I decided to go with storing count explicitly to deal with 0 rows case because in the next implementation for Avg, we need to store that count anyway. So it is better to implement it here and then reuse it for Avg.

And the implementation of normal AVG is actually similar to what we have here: store count explicitly and check if count==0 then output NULL.